### ISSUE TITLE
pulsar-client-admin-api has excessive transient dependencies

### ISSUE THREAD

**Issue by lhotari** [apache/pulsar#9640]
**Describe the bug**

It seems that the intention of `pulsar-client-admin-api` introduced in #9246 has been to provide a library that has minimal transient dependencies in the similar way as `pulsar-client-api`. However this is not the case. `pulsar-client-admin-api` has a lot of transient dependencies.

**To Reproduce**

```
$ mvn -pl pulsar-client-api dependency:tree -Dscope=runtime
.
.
.
[INFO] --- maven-dependency-plugin:2.10:tree (default-cli) @ pulsar-client-api ---
[INFO] org.apache.pulsar:pulsar-client-api:jar:2.8.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
```

`pulsar-client-admin-api` has a lot of dependencies at runtime

```
$ mvn -pl pulsar-client-admin-api dependency:tree -Dscope=runtime
.
.
.
[INFO] --- maven-dependency-plugin:2.10:tree (default-cli) @ pulsar-client-admin-api ---
[INFO] org.apache.pulsar:pulsar-client-admin-api:jar:2.8.0-SNAPSHOT
[INFO] +- org.apache.pulsar:pulsar-common:jar:2.8.0-SNAPSHOT:compile
[INFO] |  +- org.apache.pulsar:pulsar-client-api:jar:2.8.0-SNAPSHOT:compile
[INFO] |  +- io.swagger:swagger-annotations:jar:1.6.2:compile
[INFO] |  +- org.slf4j:slf4j-api:jar:1.7.25:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.11.1:compile
[INFO] |  |  +- com.fasterxml.jackson.core:jackson-annotations:jar:2.11.1:compile
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-core:jar:2.11.1:compile
[INFO] |  +- com.google.guava:guava:jar:30.1-jre:compile
[INFO] |  |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |  |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] |  |  +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO] |  |  +- org.checkerframework:checker-qual:jar:3.5.0:compile
[INFO] |  |  +- com.google.errorprone:error_prone_annotations:jar:2.3.4:compile
[INFO] |  |  \- com.google.j2objc:j2objc-annotations:jar:1.3:compile
[INFO] |  +- io.netty:netty-handler:jar:4.1.51.Final:compile
[INFO] |  |  +- io.netty:netty-common:jar:4.1.51.Final:compile
[INFO] |  |  +- io.netty:netty-resolver:jar:4.1.51.Final:compile
[INFO] |  |  +- io.netty:netty-buffer:jar:4.1.51.Final:compile
[INFO] |  |  +- io.netty:netty-transport:jar:4.1.51.Final:compile
[INFO] |  |  \- io.netty:netty-codec:jar:4.1.51.Final:compile
[INFO] |  +- io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.51.Final:compile
[INFO] |  |  \- io.netty:netty-transport-native-unix-common:jar:4.1.51.Final:compile
[INFO] |  +- io.netty:netty-transport-native-unix-common:jar:linux-x86_64:4.1.51.Final:compile
[INFO] |  +- org.apache.bookkeeper:bookkeeper-common-allocator:jar:4.12.1:compile
[INFO] |  +- io.airlift:aircompressor:jar:0.16:compile
[INFO] |  +- org.apache.bookkeeper:circe-checksum:jar:4.12.1:compile
[INFO] |  +- io.netty:netty-tcnative-boringssl-static:jar:2.0.33.Final:compile
[INFO] |  +- io.netty:netty-codec-haproxy:jar:4.1.51.Final:compile
[INFO] |  +- org.eclipse.jetty:jetty-util:jar:9.4.35.v20201120:compile
[INFO] |  +- org.apache.commons:commons-lang3:jar:3.6:compile
[INFO] |  +- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.11.1:compile
[INFO] |  |  \- org.yaml:snakeyaml:jar:1.26:compile
[INFO] |  +- javax.ws.rs:javax.ws.rs-api:jar:2.1:compile
[INFO] |  \- commons-io:commons-io:jar:2.5:compile
[INFO] +- org.apache.pulsar:pulsar-client-original:jar:2.8.0-SNAPSHOT:compile
[INFO] |  +- org.apache.pulsar:pulsar-transaction-common:jar:2.8.0-SNAPSHOT:compile
[INFO] |  |  +- com.google.protobuf:protobuf-java:jar:3.11.4:compile
[INFO] |  |  \- com.google.protobuf:protobuf-java-util:jar:3.11.4:compile
[INFO] |  +- org.apache.pulsar:bouncy-castle-bc:jar:pkg:2.8.0-SNAPSHOT:compile
[INFO] |  |  +- org.bouncycastle:bcpkix-jdk15on:jar:1.68:compile
[INFO] |  |  |  \- org.bouncycastle:bcprov-jdk15on:jar:1.68:compile
[INFO] |  |  \- org.bouncycastle:bcprov-ext-jdk15on:jar:1.68:compile
[INFO] |  +- io.netty:netty-codec-http:jar:4.1.51.Final:compile
[INFO] |  +- io.netty:netty-resolver-dns:jar:4.1.51.Final:compile
[INFO] |  |  \- io.netty:netty-codec-dns:jar:4.1.51.Final:compile
[INFO] |  +- org.asynchttpclient:async-http-client:jar:2.12.1:compile
[INFO] |  |  +- org.asynchttpclient:async-http-client-netty-utils:jar:2.12.1:compile
[INFO] |  |  +- org.reactivestreams:reactive-streams:jar:1.0.3:compile
[INFO] |  |  +- com.typesafe.netty:netty-reactive-streams:jar:2.0.4:compile
[INFO] |  |  \- com.sun.activation:javax.activation:jar:1.2.0:compile
[INFO] |  +- commons-codec:commons-codec:jar:1.10:compile
[INFO] |  +- com.yahoo.datasketches:sketches-core:jar:0.8.3:compile
[INFO] |  |  \- com.yahoo.datasketches:memory:jar:0.8.3:compile
[INFO] |  +- com.google.code.gson:gson:jar:2.8.6:compile
[INFO] |  +- org.apache.avro:avro:jar:1.9.1:compile
[INFO] |  |  \- org.apache.commons:commons-compress:jar:1.19:compile
[INFO] |  +- org.apache.avro:avro-protobuf:jar:1.9.1:compile
[INFO] |  +- com.fasterxml.jackson.module:jackson-module-jsonSchema:jar:2.11.1:compile
[INFO] |  |  \- javax.validation:validation-api:jar:1.1.0.Final:compile
[INFO] |  \- net.jcip:jcip-annotations:jar:1.0:compile
[INFO] \- org.apache.pulsar:pulsar-package-core:jar:2.8.0-SNAPSHOT:compile
[INFO] ------------------------------------------------------------------------
```

**Expected behavior**

It is expected that `pulsar-client-admin-api` has minimal transient dependences (0 or few).

**Comment by jerrypeng** [comment]
@lhotari thanks for filling this issue!

**Comment by lhotari** [comment]
@sijie @freeznet What is the status of the fix? The release date of 2.8.0 is pretty close and this is a blocker.
